src/DicomMessage.js

```diff
@@ -209,18 +209,61 @@ class DicomMessage {
             throw new Error("Invalid DICOM file, expected header is missing");
         }
 
+        // save position before reading first tag
+        var metaStartPos = stream.offset;
+
+        // read the first tag to check if it's the meta length tag
         var el = DicomMessage._readTag(stream, useSyntax);
+
         if (el.tag.toCleanString() !== "00020000") {
-            throw new Error(
-                "Invalid DICOM file, meta length tag is malformed or not present."
-            );
-        }
+            // meta length tag is missing
+            if (!options.ignoreErrors) {
+                throw new Error(
+                    "Invalid DICOM file, meta length tag is malformed or not present."
+                );
+            }
+
+            // read meta header elements sequentially
+            var metaHeader = {};
+            // reset stream to the position where we started reading tags
+            stream.offset = metaStartPos;
+
+            // read meta header elements until we encounter a tag outside the 0002 group
+            while (!stream.end()) {
```

Comment:
Can you use the _read method with an until tag of 00030000 ? I'd rather re-use that as much as possible

Reply:
@wayfarer3130 - Thanks, I implemented your request.

Note: The existing _read method was reading all tags if the untilTag was not found, so I added an optional "stopOnGreaterTag" parameter that causes it to stop when it encounters a tag greater than untilTag. This allows us to read only the meta header (group 0002) by using untilTag: "00030000" with stopOnGreaterTag: true.